sql/schema_changer.go

```diff
@@ -367,16 +367,16 @@ func (sc *SchemaChanger) purgeMutations(lease *TableDescriptor_SchemaChangeLease
 	// changers keep attempting to apply and purge mutations.
 	// This is a theoretical problem at this stage (2015/12).
 	if pErr := sc.applyMutations(lease); pErr != nil {
```

Comment:
We can't do that now as we have the following flow: 

```
   applyMutations
-> SchemaChanger.exec
-> Executor.execStmts
```

and `execStmts` calls `pErr.GetDetail()`. I think we can make `applyMutations` return `error` if we have an SQL-specific error conversion as you mentioned.